### PR TITLE
PLT-632: Added node maintenance status for all clouds

### PR DESCRIPTION
**Pull request by SivaanandM**

### CODE REVIEW

File: spectrocloud/schemas/cluster_node.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+func NodeSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"node_id": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "The node_id of the node.",
```
Comment (nikchern) [inline spectrocloud/schemas/cluster_node.go:17]: Please provide an example how the node id may look like or where can be taken from i.e. API reference.

File: spectrocloud/schemas/cluster_node.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+				},
+				"action": {
+					Type:         schema.TypeString,
+					Required:     true,
+					Description:  "The action to perform on the node",
```
Comment (nikchern) [inline spectrocloud/schemas/cluster_node.go:22]:
```suggestion
					Description:  "The action to perform on the node. Valid values are: `cordon`, `uncordon`.",
```

**Review by nikchern — CHANGES_REQUESTED** [review]

File: spectrocloud/resource_cluster_vsphere_test.go [hunk 3]
```diff
@@ -84,6 +84,7 @@ func prepareClusterVsphereTestData() *schema.ResourceData {
 		"count":                   1,
 		"placement":               masterPlacement,
 		"instance_type":           masterInstance,
+		"node":                    []interface{}{},
```
Comment (nikchern) [inline spectrocloud/resource_cluster_vsphere_test.go:87]: Is there any assert that is actually checking the value?

**Review by nikchern — APPROVED** [review]

File: spectrocloud/resource_cluster_vsphere_test.go (hunk 3 above)
Comment (SivaanandM) [reply spectrocloud/resource_cluster_vsphere_test.go:87]: Basically node will fall under day2 operation we cant mimc d.HasChange("machine_pool")

**Pull request merged**